gadget/gadget.go

```diff
@@ -326,6 +327,77 @@ func LoadDiskVolumesDeviceTraits(dir string) (map[string]DiskVolumeDeviceTraits,
 	return mapping, nil
 }
 
+// AllDiskVolumeDeviceTraits takes a mapping of volume name to LaidOutVolume and
+// produces a map of volume name to DiskVolumeDeviceTraits. Since doing so uses
+// DiskVolumeDeviceTraitsForDevice, it will also validate that disk devices
+// identified for the laid out volume are compatible and matching before
+// returning.
+func AllDiskVolumeDeviceTraits(allLaidOutVols map[string]*LaidOutVolume, optsPerVolume map[string]*DiskVolumeValidationOpts) (map[string]DiskVolumeDeviceTraits, error) {
```

Comment:
what is the plan concretely to fill optsPerVolume ?  I just wondering if global options make sense instead, so that's why the question

Reply:
The reason it is `optsPerVolume` and not one set of the same Options for all volumes is that in a followup PR, I need to add another option which will indicated expected encrypted structures on specific volumes so that EnsureLayoutCompatibility will change its logic slightly to check that i.e. a normal TPM/LUKS encrypted structure has `Filesystem == "crypto_LUKS"` instead of `ext4` as the gadget.yaml says. That specific encrypted option needs to be set only for specific volumes, hence the map of options per volume